app/Http/Controllers/MasjidProfileController.php

```diff
@@ -39,4 +40,32 @@ public function update(Request $request): RedirectResponse
 
         return redirect()->route('masjid_profile.show');
     }
+
+    public function image(Request $request)
+    {
+        $this->authorize('edit_masjid_profile');
+
+        $validatedPayload = $request->validate([
+            'image' => 'required'
+        ]);
+
+        if (!base64_decode($validatedPayload['image'])) {
+            return response()->json([
+                'message' => 'Image not found'
+            ]);
+        }
+
+        $image_parts      = explode(";base64,", $request->image);
+        $image_base64     = base64_decode($image_parts[1]);
+        $image_name       = uniqid() . '.webp';
+        $image_full_path  = 'public/' . $image_name;
+
+        Storage::put($image_full_path, $image_base64);
+        Setting::set('masjid_image', $image_name);
```

Comment:
Sebelum kita proses ke sini, sebaiknya kita:
1. Sebelum simpan ke storage dan database, lakukan cek dan hapus dulu image logo yang lama.
2. Untuk setting key, kita lebih explisit bahwa yang kita simpan adalah file path logo, misal: `masjid_logo_path`

```suggestion
       // Cek dulu apakah sudah ada image yang sudah ada
       // Jika sudah ada file logo yang lama, kita hapus dulu.
       
        Storage::put($image_full_path, $image_base64);
        Setting::set('masjid_logo_path', $image_name);
```